docs/docs/50-user-guide/60-reference-docs/30-promotion-steps/jira.md

```diff
@@ -0,0 +1,651 @@
+
+This example deletes a specific comment from a Jira issue.
+
+```yaml
+steps:
+  - uses: jira
+    config:
+      credentials:
+        secretName: jira-credentials
+      deleteComment:
+        issueKey: "${{ freightMetadata(ctx.targetFreight.name, 'jira-issue-key') }}"
+        commentID: "${{ outputs['previous-comment-step'].commentID }}"
+```
+
+## Status Tracking
+
+### Wait for Status
+
+Waits for a Jira issue to reach a specific status before proceeding.
+
+#### Configuration
+
+| Name                           | Type     | Required | Description                                                    |
+| ------------------------------ | -------- | -------- | -------------------------------------------------------------- |
+| `waitForStatus.issueKey`       | `string` | Y        | The Jira Issue Key (e.g., EXT-123).                            |
+| `waitForStatus.expectedStatus` | `string` | Y        | The expected status to wait for (e.g., 'IN PROGRESS', 'DONE'). |
+
+#### Output
+
+This step does not produce any output values.
+
+#### Example
+
+This example waits for a change request issue to be approved before proceeding with deployment.
+
+```yaml
+steps:
+  - uses: jira
+    config:
+      credentials:
+        secretName: jira-credentials
+      waitForStatus:
+        issueKey: "${{ freightMetadata(ctx.targetFreight.name, 'change-request-key') }}"
+        expectedStatus: "Approved"
+  # Deployment steps continue after approval...
+  - uses: helm-template
+    config:
+      path: ./charts
+      vars:
+        imageTag: "${{ imageFrom(vars.imageRepo).Tag }}"
+        environment: "${{ ctx.stage.name }}"
+```
+
+## Multi-Stage Workflow
+
+The Jira promotion step automatically stores created issue keys in the Freight metadata, allowing subsequent stages to reference the same issue. This enables tracking a single Jira issue across multiple promotion stages.
+
+### Accessing Issue Keys from Freight Metadata
+
+Use the `freightMetadata()` template function to retrieve issue keys stored by previous stages:
+
+```yaml
+# Access the default issue key
+issueKey: ${{ freightMetadata(ctx.targetFreight.name, 'jira-issue-key') }}
+
+# Access a custom issue key (when issueAlias was used during creation)
+issueKey: ${{ freightMetadata(ctx.targetFreight.name, 'my-custom-alias') }}
+```
+
+### Example
+
+This comprehensive example demonstrates how to use the Jira promotion step across multiple stages in a promotion pipeline, tracking a single issue from creation through production deployment:
+
+```yaml
+---
+apiVersion: kargo.akuity.io/v1alpha1
+kind: Stage
+metadata:
+  name: test
+  namespace: kargo-proj
+spec:
+  requestedFreight:
+    - origin:
+        kind: Warehouse
+        name: nginx
+      sources:
+        direct: true
+  promotionTemplate:
+    spec:
+      vars:
+        - name: imageRepo
+          value: public.ecr.aws/nginx/nginx
+      steps:
+        # Create initial deployment ticket
+        - as: create-deployment-ticket
```

Comment:
I'm wondering what happens if/when Freight is _re_-promoted to this Stage. We get a branch new issue and the ID of that new issue replaces the ID of the old issue in the metadata?

I think there are all kinds of weird things that might happen.

1. Promote to test: New issue
1. Promote to stage: Update issue
1. Promote _something else_ to test and decide you don't like it
1. Promote test back to the older Freight: New issue? Do we now, essentially lose track of all the successful promotions that already happened downstream from test?

Reply:
Looking at the implementation, I think perhaps we ought to make "create issue" fall back on adding a comment to the existing issue instead if the Freight's metadata already has an issue under the key... but this would change how we might wish to deal with failures in promotions to test. Instead of cleaning up by _deleting_ the issue, we could clean up by adding a comment. Other stages that update the issue with a comment, likewise, could clean up by adding an additional comment (instead of one they already created)...

i.e. If we start to look at a single Jira ticket as a write-only ledger of activity recorded in the comments, non-linear flows will become a bit easier.

Reply:
> Not sure how to exactly frame this w.r.t to kargo for us but I would prefer to not include any custom handling in the low level step action itself and let user figure out a way they prefer to deal with this.

Ya... I agree with us not getting overly opinionated about how to deal with this sort of thing. Let users decide. At the same time, I think we should _also_ avoid requiring too much boilerplate from users that must be _just so_ in order to escape the unconditional new ticket footgun.

But if we're calling the feature beta, I think we'll learn more about realistic usage patterns by using it and by observing what kind of things other users end up doing with it that we may not even have imagined.

I think we could let this go for now.